docs: developer incentives, register cosmos coins

adisaran64 opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

- closes: [ENG-572](https://linear.app/evmos/issue/ENG-572/create-user-incentives-step-by-step-guide), [ENG-576](https://linear.app/evmos/issue/ENG-576/devs-dont-know-how-to-register-a-cosmos-coin-with-erc20-module)
- for ENG-576, added Cosmos Coin considerations to the doc about interoperable ERC-20s, instead of copying the file and making minor changes

______

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.

PR review checkboxes:

I have...

- [ ] added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link in the PR description to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all required CI checks have passed

Code maintenance:

I have...

- [ ] written unit and integration [tests](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#testing)
- [ ] added relevant [`godoc`](https://go.dev/blog/godoc) and [code comments](https://blog.jbowen.dev/2019/09/the-magic-of-go-comments/).
- [ ] updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)

______

### Reviewers Checklist

**All** items are required. Please add a note if the item is not applicable and please add your handle next to the items reviewed if you only reviewed selected items.

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] confirmed that this PR does not change production code

danburck commented on docs/developers/guides/coin_token_registration.md:
```diff
@@ -13,19 +13,28 @@ Application-wise, the ERC-20 module allows DeFi protocols to seamlessly integrat
+3. [The On-Chain Registration Proposal](#the-on-chain-registration-proposal)
 
-## Drafting the ERC-20 Proposal
+:::tip
```
Remove tip formatting. These notes should be used for short explanations instead.

danburck commented on docs/developers/guides/coin_token_registration.md:
```diff
@@ -52,9 +61,9 @@ The following topics must be addressed when drafting an ERC-20 Proposal:
 Remember to provide links to the relevant [Commonwealth Evmos community](https://commonwealth.im/evmos) discussions concerning your proposal, as well as the [proposal on testnet](#submit-the-proposal-to-the-testnet).
 
-## Submitting the ERC-20 Proposal
+## Submitting the Registration Proposal
```
The cli command in this section is wrong its missing `register-coin` and `register-erc20`, e.g. ` evmosd tx gov submit-proposal register-erc20`

danburck reviewed (commented):
I'd split up the Cosmos Coin and ERC20 token guides, either in two guides or one well-structured one.  Users and devs have been confused by the difference between the two.

Also, why do you include different details on proposals and node configuration in the incentives and the erc20 guide? I recommend focussing on short explanations using text over bullet points. For reference have a look at the `dApp Store Contract Registration`

adisaran64 commented:
@danburck reworked this PR with the following:

- split up the cosmos coin and erc20 registration guides, just too clunky to integrate into one file
- avoided use of bullet points and tips, opting for short text blurbs instead, rewrote drafting requirements to numbered list
- included only one tip at beginning of both files for users struggling to understand difference between the erc20/cosmos coin proposals
- corrected cli commands in erc20 and cosmos coin files
- fixed incentives guide to look like the revamped erc20/cosmos coin guide

did some basic ordering as well, planning to fix structure of the guides with the restructuring PR. 
let me know if any adjustments need to be made.

fedekunze commented on docs/developers/guides/cosmos_coin_registration.md:
```diff
@@ -0,0 +1,131 @@
+Still confused? Learn more about the differences [here](../../../x/erc20/spec/01_concepts.md).
+:::
+
+Learn how to register native Cosmos Coins through [Evmos Governance](../../users/governance/overview.md). {synopsis}
```
synopsis should come before the `::: tip`

fedekunze commented on docs/developers/guides/incentives_registration.md:
````diff
@@ -0,0 +1,163 @@
+
+To enable telemetrics, set `telemetry.enabled = true` in the `app.toml` config file of the node. The Cosmos SDK currently supports enabling in-memory and [Prometheus](https://prometheus.io/) telemetry sinks. The in-memory sink is always attached (when telemetry is enabled) with a ten second interval and one minute retention. This means that metrics will be aggregated over ten seconds, and metrics ...
+
+### Emitting & Collecting Metrics
+
+If telemetry is enabled via configuration, a single global metrics collector is exposed via the [`go-metrics`](https://github.com/armon/go-metrics) library. This allows for emitting and collecting metrics through a simple [API](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-rc1/telemetry/wrapper.go). For example:
+
+```go
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
+  defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+
+  // ...
+}
+```
+
+Developers may use the [`telemetry`](https://docs.evmos.org/protocol/telemetry.html) package directly, which provides wrappers around metric APIs that include adding useful labels, or they must use the `go-metrics` library directly. It is preferable to add as much context and adequate dimensionality to metrics as possible, so the `telemetry` package is advised.
+
+Regardless of the package or method used, the Cosmos SDK supports `guages`, `summaries`, and `counters` as metric types.
````
this is not relevant for a developer since they won't modify the golang code, they just need to know how to use the telemetry

danburck commented on docs/developers/guides/erc20_registration.md:
````diff
@@ -66,27 +70,24 @@ writing markdown files.
-evmosd tx gov submit-proposal --proposal=<path_to_json>
+evmosd tx gov submit-proposal register-erc20 --proposal=<path_to_json>
 ```
 
````
provide a proposal to the json file

adisaran64 replied:
Link to the process of creating a proposal is above the code block, think it's redundant to have it here as well

danburck commented on docs/developers/guides/erc20_registration.md:
````diff
@@ -66,27 +70,24 @@ writing markdown files.
 
 ```bash
-evmosd tx gov submit-proposal --proposal=<path_to_json>
+evmosd tx gov submit-proposal register-erc20 --proposal=<path_to_json>
````
nit, use consistent format for the path, you used `<path/to/metadata.json>` above

danburck commented on docs/developers/guides/incentives_registration.md:
```diff
@@ -0,0 +1,163 @@
+
+Learn how to register smart contract incentives through [Evmos Governance](../../users/governance/overview.md). {synopsis}
+
+Smart contracts can register incentives through governance. After incentives are registered for a smart contract, users that interact with the contract (eg. through submission of transactions) will have rewards distributed to them. These rewards are essentially cash-back for the gas spent on the initial transactions.
```
```suggestion
An incentive for a smart contract can be registered through governance to encourage more interaction with that contract. Users who interact with a registered contract (eg. through submission of transactions) will have rewards distributed to them. These rewards are essentially cash-back for the gas spent on the initial transactions.
```

danburck commented on docs/developers/guides/incentives_registration.md:
```diff
@@ -0,0 +1,163 @@
+3. [The On-Chain Contract Incentives Proposal](#the-on-chain-contract-incentives-proposal)
+
+This document also discusses [incentives analysis through telemetry](#incentives-analysis-through-telemetry).
+
```
Add like to spec for more info

danburck reviewed (changes requested):
@adisaran64 Great! This flows much better. Just left a few more comments, then we're good to go

adisaran64 commented:
@danburck fixed issues, ready for review

fedekunze commented:
@danburck can you re-review?
